### PR TITLE
[uchardet] Add new port

### PR DESCRIPTION
**Pull request by myd7349**
Related issue: #8088 .

### CODE REVIEW

**Comment by vicroms** [comment]
UWP failures

```
CMake Error at CMakeLists.txt:42 (string):
  string no output variable specified
```

**Comment by myd7349** [comment]
> 
> 
> UWP failures
> 
> ```
> CMake Error at CMakeLists.txt:42 (string):
>   string no output variable specified
> ```

@vicroms Thanks for your kindly help!

File: ports/uchardet/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,59 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/uchardet/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/uchardet/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    file(REMOVE ${CURRENT_PACKAGES_DIR}/bin/uchardet${VCPKG_TARGET_EXECUTABLE_SUFFIX})
+endif()
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/debug/share
+    ${CURRENT_PACKAGES_DIR}/share/man
+)
+
+configure_file(${SOURCE_PATH}/COPYING ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
Comment (LilyWangL) [inline ports/uchardet/portfile.cmake]: Can you please use command `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Seems build failure on uwp, can you check that?

Thanks.

**Review by LilyWangL — APPROVED** [review]

**Comment by strega-nil** [comment]
Alright, looks good! Thanks again @myd7349 :)

**Pull request merged**